frontend/src/upload.js

```diff
@@ -66,35 +68,44 @@ $(document).ready(function() {
       $('#page-one').hide();
       $('#file-list').hide();
       $('#upload-progress').show();
+      $('#upload-error').hide();
       $('#upload-filename').innerHTML += file.name;
       // update progress bar
       document
         .querySelector('#progress-bar')
         .style.setProperty('--progress', percentComplete + '%');
       $('#progress-text').html(`${percentComplete}%`);
     });
-    fileSender.upload().then(info => {
-      const url = info.url.trim() + `#${info.secretKey}`.trim();
-      $('#link').attr('value', url);
-      const fileData = {
-        name: file.name,
-        fileId: info.fileId,
-        url: info.url,
-        secretKey: info.secretKey,
-        deleteToken: info.deleteToken,
-        creationDate: new Date(),
-        expiry: expiration
-      };
-      localStorage.setItem(info.fileId, JSON.stringify(fileData));
-
-      $('#page-one').hide();
-      $('#file-list').hide();
-      $('#upload-progress').hide();
-      $('#share-link').show();
-
-      populateFileList(JSON.stringify(fileData));
-      notify('Your upload has finished.');
-    });
+    fileSender
+      .upload()
+      .then(info => {
+        const url = info.url.trim() + `#${info.secretKey}`.trim();
+        $('#link').attr('value', url);
+        const fileData = {
+          name: file.name,
+          fileId: info.fileId,
+          url: info.url,
+          secretKey: info.secretKey,
+          deleteToken: info.deleteToken,
+          creationDate: new Date(),
+          expiry: expiration
+        };
+        localStorage.setItem(info.fileId, JSON.stringify(fileData));
+
+        $('#page-one').hide();
+        $('#file-list').hide();
+        $('#upload-progress').hide();
+        $('#share-link').show();
+        $('#upload-error').hide();
+
+        populateFileList(JSON.stringify(fileData));
+        notify('Your upload has finished.');
+      })
+      .catch(err => {
+        console.log('Upload error name: ' + err);
```

Comment:
I think that only sending the error to Raven here is sufficient? This will eliminate any duplicates, since errors from the `reader.onerror` triggers (https://github.com/mozilla/something-awesome/pull/168#discussion-diff-126022292R54) are always caught here.